src/initialization.jl

```diff
@@ -72,10 +76,17 @@ function __init__()
     if isdefined(Base, :active_repl_backend) && !isnothing(Base.active_repl_backend)
         push!(Base.active_repl_backend.ast_transforms, synchronize_metal_tasks)
     end
+
+    # Start async warmup to reduce first-kernel JIT compilation latency.
+    # Only run with multiple threads - with a single thread, the async task would
+    # block the main thread due to Julia's cooperative task runtime.
+    return if functional() && _warmup_enabled && Threads.nthreads() > 1
+        _warmup_task[] = errormonitor(@async _warmup_compilation())
```

Comment:
```suggestion
        _warmup_task[] = errormonitor(Threads.@spawn _warmup_compilation())
```

`@async` is pinned to the same thread as parent.